src/generators/minecraft-item/MinecraftItemGenerator.res

```diff
@@ -23,13 +23,44 @@ The generator supports four standard sizes:
 
 let images: array<Generator.imageDef> = [
   {id: "Background", url: Generator.requireImage("./images/Background.png")},
+  {id: "Title", url: Generator.requireImage("./images/Title.png")},
 ]
 
-let textures: array<Generator.textureDef> = TextureVersions.allTextureDefs
+let textures: array<Generator.textureDef> = Js.Array.concat(
+  TextureVersions.allTextureDefs,
+  [
+    {
+      id: "CenterFold",
+      url: Generator.requireImage("./item-textures/CenterFold.png"),
+      standardWidth: 2,
+      standardHeight: 128,
+    },
+  ],
+)
+
+let drawItem = (id, rectangle, x, y, size, page, showFolds) => {
```

Comment:
Excellent work here. A couple of tweaks we can make:

[1] `cycleTextureOffset` is an independent function, so is good to move outside this function.

[2] Each particular item texture can be uniquely identified by the `id` (the texture "version" selected) plus the x and y coordinates inside the `rectangle` (the item's texture position).

[3] The `rectangle` variable contains the width of the original texture, which might be 16, 32, 64 etc pixels. We can use that for offset max value.

Here's some code that shows all of these changes:

```res
let cycleTextureOffset = (t, tileWidth) => {
  let t = if t === tileWidth {
    0
  } else {
    t + 1
  }
  Belt.Int.toString(t)
}

let makeRegionId = (textureId, rectangle) => {
  let (tileX, tileY, _, _) = rectangle
  textureId ++ "-" ++ Js.Int.toString(tileX) ++ "-" ++ Js.Int.toString(tileY)
}

let getTileWidth = rectangle => {
  let (_, _, tileWidth, _) = rectangle
  tileWidth
}

let drawItem = (textureId, rectangle, x, y, size, showFolds) => {
  let tileWidth = getTileWidth(rectangle)
  let regionId = makeRegionId(textureId, rectangle)

  let textureOffset =
    Generator.getSelectInputValue(regionId)->Belt.Int.fromString->Belt.Option.getWithDefault(0)

  Generator.defineRegionInput((x, y, size, size), () => {
    Generator.setSelectInputValue(regionId, cycleTextureOffset(textureOffset, tileWidth))
  })

  let offset = textureOffset * size / tileWidth

  Generator.drawTexture(textureId, rectangle, (x + offset, y, size, size), ())

  Generator.drawTexture(
    textureId,
    rectangle,
    (x + size - offset, y, size, size),
    ~flip=#Horizontal,
    (),
  )

  if showFolds {
    Generator.drawTexture("CenterFold", (0, 0, 2, size), (x + size - 1, y, 2, size), ())
  }
}
```